test/e2eexternal/dataplane/connectivity.go

```diff
@@ -0,0 +1,335 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dataplane
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var _ = Describe("[external][dataplane] Connectivity", func() {
+	f := framework.NewFramework("ext-dataplane")
+
+	It("Should be able to connect from external app to pod in cluster", func() {
+		testExternalConnectivity(f)
+	})
+})
+
+const (
+	extAppName = "ext-app"
+	extNetName = "pseudo-ext"
+
+	testImage         = "registry.access.redhat.com/ubi7/ubi:latest"
+	testContainerName = "ext-test-container"
+	testLabelkey      = "ext-test"
+)
+
+var (
+	podNameBase             = "ext-test-pod"
+	svcName                 = "ext-test-svc"
+	simpleHTTPServerCommand = []string{"python", "-m", "SimpleHTTPServer", "80"}
+	testLabel               = map[string]string{testLabelkey: podNameBase}
+)
+
+func testExternalConnectivity(f *framework.Framework) {
+	for idx, cs := range framework.KubeClients {
+		By(fmt.Sprintf("Creating a pod in cluster %q", framework.TestContext.ClusterIDs[idx]))
+		var err error
+
+		pod := makePodSpec(podNameBase, testLabel, testContainerName, testImage, simpleHTTPServerCommand)
```

Comment:
I know it's a little bit more complex, but can we add this type of pod to the framework:

https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/network_pods.go#L84

So then we can use it from the framework here (and may be make it useful in other places too?)